--- SSLeay.xs	2020-04-30 14:48:03.897259294 +0000
+++ SSLeay.xs	2020-04-30 14:51:14.193744640 +0000
@@ -149,15 +149,15 @@
             ctx = SSL_CTX_new(SSLv23_client_method());
         }
         else if(ssl_version == 3) {
-            ctx = SSL_CTX_new(SSLv3_client_method());
+            ctx = SSL_CTX_new(TLS_client_method());
         }
         else {
-#ifndef OPENSSL_NO_SSL2
+#if !defined OPENSSL_NO_SSL2 && OPENSSL_VERSION_NUMBER < 0x10100000L
             /* v2 is the default */
             ctx = SSL_CTX_new(SSLv2_client_method());
 #else
             /* v3 is the default */
-            ctx = SSL_CTX_new(SSLv3_client_method());
+            ctx = SSL_CTX_new(TLS_client_method());
 #endif
         }
 
