diff -purd a/Inotify2.pm b/Inotify2.pm
--- a/Inotify2.pm	2018-10-25 03:19:02.000000000 +0200
+++ b/Inotify2.pm	2018-10-25 12:45:08.380829396 +0200
@@ -75,7 +75,8 @@ package Linux::Inotify2;
 
 use Scalar::Util ();
 
-use common::sense;
+use strict;
+use warnings;
 
 use Exporter qw(import);
 
@@ -430,6 +431,8 @@ sub fullname {
 }
 
 for my $name (@Linux::Inotify2::EXPORT) {
+   no strict "refs";
+
    my $mask = &{"Linux::Inotify2::$name"};
 
    *$name = sub { $_[0]{mask} & $mask };
diff -purd a/Makefile.PL b/Makefile.PL
--- a/Makefile.PL	2009-09-21 10:08:20.000000000 +0200
+++ b/Makefile.PL	2018-10-25 12:45:08.380829396 +0200
@@ -8,9 +8,6 @@ my $mm = MM->new({
 	            },
     NAME         => "Linux::Inotify2",
     VERSION_FROM => "Inotify2.pm",
-    PREREQ_PM    => {
-       common::sense => 0,
-    },
 });
 
 $mm->flush;
diff -purd a/META.json b/META.json
--- a/META.json	2018-10-25 03:19:27.000000000 +0200
+++ b/META.json	2018-10-25 12:45:50.945959575 +0200
@@ -29,11 +29,6 @@
          "requires" : {
             "ExtUtils::MakeMaker" : "0"
          }
-      },
-      "runtime" : {
-         "requires" : {
-            "common::sense" : "0"
-         }
       }
    },
    "release_status" : "stable",
diff -purd a/META.yml b/META.yml
--- a/META.yml	2018-10-25 03:19:27.000000000 +0200
+++ b/META.yml	2018-10-25 12:46:02.610269294 +0200
@@ -17,7 +17,5 @@ no_index:
   directory:
     - t
     - inc
-requires:
-  common::sense: '0'
 version: '2.0'
 x_serialization_backend: 'CPAN::Meta::YAML version 0.012'
