diff --git a/Changes b/Changes
index 45727a3..0780434 100644
--- a/Changes
+++ b/Changes
@@ -21,6 +21,9 @@ Revision history for Perl extension Net:
 	    processes a ClientHello message.
 	  - Many more libcrypto/libssl constants and functions are now exposed; see the
 	    release notes for the 1.93 developer releases for a full list.
+	  - Adjust test 32_x509_get_cert_info.t to match formatting
+	    changes in OpenSSL 3.4.0 and 3.4.1. Thanks to Sebastian
+	    Andrzej Siewior for the patches.
 
 1.93_05 2024-01-06
 	- Remove support for automatic detection of libssl/libcrypto via pkg-config
 
 1.94 2024-01-08
 	- New stable release incorporating all changes from developer releases 1.93_01
diff --git a/t/local/32_x509_get_cert_info.t b/t/local/32_x509_get_cert_info.t
index 0f7e2d5..08316bf 100644
--- a/t/local/32_x509_get_cert_info.t
+++ b/t/local/32_x509_get_cert_info.t
@@ -188,6 +188,10 @@ for my $f (keys (%$dump)) {
                   ) {
                       $ext_data =~ s{(othername:) [^, ]+}{$1<unsupported>}g;
                   }
+                  # Starting with 3.4.0 the double colon in emailAddress has been removed.
+                  if (Net::SSLeay::SSLeay >= 0x30400000) {
+                      $ext_data =~ s{emailAddress::}{emailAddress:};
+                  }
               }
               elsif ( $nid == 89 ) {
                   # The output formatting for certificate policies has a
@@ -214,6 +218,9 @@ for my $f (keys (%$dump)) {
                       # OpenSSL 1.0.0 to 1.1.1:
                       $ext_data =~ s{(Full Name:\n  )}{\n$1}g;
                       $ext_data .= "\n";
+                  } elsif ( Net::SSLeay::SSLeay >  0x3040000f ) {
+                      $ext_data =~ s{(\nFull Name:)}{\n$1}g;
+                      $ext_data .= "\n";
                   }
               }
               elsif ( $nid == 126 ) {
--- Changes.orig	2024-01-08 02:17:02.000000000 +0100
+++ Changes	2025-06-25 11:09:10.129980580 +0200
