diff -purd String-Diff-0.07-0/t/02_diff_fully-pp.t String-Diff-0.07-1/t/02_diff_fully-pp.t
--- String-Diff-0.07-0/t/02_diff_fully-pp.t	2015-08-23 10:58:44.000000000 +0200
+++ String-Diff-0.07-1/t/02_diff_fully-pp.t	2018-12-10 18:14:45.707113206 +0100
@@ -30,7 +30,7 @@ new: perl
 --- data1
 -
  - ['-', 'perl']
--
+- []
 --- data2
 old: perl
 new: 
diff -purd String-Diff-0.07-0/t/02_diff_fully.t String-Diff-0.07-1/t/02_diff_fully.t
--- String-Diff-0.07-0/t/02_diff_fully.t	2015-08-23 10:58:44.000000000 +0200
+++ String-Diff-0.07-1/t/02_diff_fully.t	2018-12-10 18:14:45.707113206 +0100
@@ -29,7 +29,7 @@ new: perl
 --- data1
 -
  - ['-', 'perl']
--
+- []
 --- data2
 old: perl
 new: 
